demos/supabase-anonymous-auth/lib/widgets/status_app_bar.dart

```diff
@@ -39,22 +40,48 @@ class _StatusAppBarState extends State<StatusAppBar> {
 
   @override
   Widget build(BuildContext context) {
-    const connectedIcon = IconButton(
-      icon: Icon(Icons.wifi),
-      tooltip: 'Connected',
-      onPressed: null,
-    );
-    const disconnectedIcon = IconButton(
-      icon: Icon(Icons.wifi_off),
-      tooltip: 'Not connected',
-      onPressed: null,
-    );
+    final statusIcon = _getStatusIcon(_connectionState);
 
     return AppBar(
       title: Text(widget.title),
       actions: <Widget>[
-        _connectionState.connected ? connectedIcon : disconnectedIcon
+        statusIcon,
+        // Make some space for the "Debug" banner, so that the status
+        // icon isn't hidden
+        if (kDebugMode) _makeIcon('Debug mode', Icons.developer_mode),
       ],
     );
   }
 }
+
+Widget _makeIcon(String text, IconData icon) {
+  return Tooltip(
+      message: text,
+      child: SizedBox(width: 40, height: null, child: Icon(icon, size: 24)));
+}
+
+Widget _getStatusIcon(SyncStatus status) {
```

Comment:
I think a switch statement would be nicer here

Reply:
A switch statement can't quite handle all the conditions used here - at least not without being more verbose.